code/components/jomjol_wlan/read_wlanini.cpp

```diff
@@ -177,9 +191,23 @@ void LoadWlanFromFile(std::string fn, char *&_ssid, char *&_password, char *&_ho
     }
     else
         _dns = NULL;
-}
 
+    if (http_username.length() > 0)
+    {
+        _http_username = new char[http_username.length() + 1];
```

Comment:
Now this part is pretty horrible. I understand that you're just extending what was already there. But dealing with raw strings when we have `std::string` available just doesn't make any sense. Who takes care of freeing that memory? Is it ever freed at all? In any case, passing references to `std::string` seems to be much better.